lib/capstone_project.ex

```diff
@@ -3,16 +3,83 @@ defmodule CapstoneProject do
   Documentation for `CapstoneProject`.
   """
 
+  @type rational :: {integer, integer}
+
+  @doc """
+  Add two rational numbers
+  """
+  @spec add(a :: rational, b :: rational) :: rational
+  def add(a, b) do
+    {{a1, b1}, {a2, b2}} = {a, b}
+    {(a1 * b2 + a2 * b1), b1 * b2}
+    |> reduce()
```

Comment:
```suggestion
    reduce({(a1 * b2 + a2 * b1), b1 * b2})
```

Один вызов функции ⇒ пайп не нужен.